### PR TITLE
immediately validate input to courtOrderPoa

### PR DESCRIPTION
**Pull request by jonathan-longe**
*Issue #:* /bcgov/entity14964

Currently the CourtOrderPoa component does not immediately report an invalid state like other shared components do.  This causes validation errors in the courtOrderPoa component to be ignored on the initial form submission.

This PR removes the autoValidation prop and changes the logic so validation errors are immediately visible after input.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/court-order-poa/CourtOrderPoa.vue [hunk 1]
```diff
@@ -45,9 +45,6 @@ export default class CourtOrderPoa extends Vue {
     courtNumRef: FormIF
   }
 
-  /** Prompt the validations. Used for global validations. */
-  @Prop({ default: false }) readonly autoValidation!: boolean
```
Comment (severinbeauvais) [inline src/components/court-order-poa/CourtOrderPoa.vue:49]: Good riddance!

**Review by cameron-eyds — APPROVED** [review]
LGTM!

File: src/components/court-order-poa/CourtOrderPoa.vue [hunk 2]
```diff
@@ -87,24 +84,21 @@ export default class CourtOrderPoa extends Vue {
...
+    // Apply TextField rules
+    this.courtOrderNumRules = [
+      (v: string) => (!v || !/^\s/g.test(v)) || 'Invalid spaces', // leading spaces
+      (v: string) => (!v || !/\s$/g.test(v)) || 'Invalid spaces', // trailing spaces
+      (v: string) => (!v || !(v.length < 5)) || 'Court order number is invalid',
+      (v: string) => (!v || !(v.length > 20)) || 'Court order number is invalid'
+    ]
+    if (this.courtOrderNumberRequired || this.planOfArrangement) {
+      this.courtOrderNumRules.push((v: string) => !!v || 'A Court Order number is required')
+    }
```
Comment (severinbeauvais) [inline src/components/court-order-poa/CourtOrderPoa.vue:100]: Could these rules be static (readonly const properties)?

File: src/components/court-order-poa/CourtOrderPoa.vue (hunk 1 above)
Comment (cameron-eyds) [reply src/components/court-order-poa/CourtOrderPoa.vue:49]: I swear there was some archaic validation requirement that spawned this blasphemous code 👀 
Good riddance indeed 😆

File: src/components/court-order-poa/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@bcrs-shared-components/court-order-poa",
-  "version": "2.1.3",
+  "version": "3.0.0",
```
Comment (severinbeauvais) [inline src/components/court-order-poa/package.json:3]: Don't change this. It is set by step 14 in the readme (which is done by a repo admin).

Comment (jonathan-longe) [reply src/components/court-order-poa/package.json:3]: Got it.  Thanks @severinbeauvais

File: src/components/court-order-poa/package.json [hunk 4]
```diff
@@ -1,11 +1,11 @@
 {
   "name": "@bcrs-shared-components/court-order-poa",
-  "version": "2.1.3",
+  "version": "2.1.4",
   "publishConfig": {
     "access": "public"
   },
   "dependencies": {
-    "@bcrs-shared-components/interfaces": "^1.0.46",
+    "@bcrs-shared-components/interfaces": "^1.0.47",
```
Comment (severinbeauvais) [inline src/components/court-order-poa/package.json:8]: This file shouldn't be updated. This, too, is done by step 14.

(The only time this file should be updated is if you add/remove a package dependency.)

File: src/components/court-order-poa/CourtOrderPoa.vue (hunk 2 above)
Comment (severinbeauvais) [reply src/components/court-order-poa/CourtOrderPoa.vue:100]: Oh wait ... that last rule is conditional, so they can't be static.

But they could be a getter.

**Review by severinbeauvais — APPROVED** [review]
LGTM

File: src/components/court-order-poa/CourtOrderPoa.vue (hunk 2 above)
Comment (severinbeauvais) [reply src/components/court-order-poa/CourtOrderPoa.vue:100]: As we all know, validation is tricky and fragile, so if this code works then I'm OK to leave it as-is.

**Pull request merged**